app/code/Magento/Ui/view/base/web/js/modal/prompt.js

```diff
@@ -152,10 +152,9 @@ define([
         /**
          * Close modal window
          */
-        closeModal: function (result) {
+        closeModal: function (event, result) {
             var value;
-
-            if (result) {
+            if (result && !(result instanceof $.Event)) {
```

Comment:
What is the purpose of the `instanceof` check?

Reply:
Actually, Initially we are passing only event so when this if condition is executed then result is taking as true so I've additional condition added that result should be instance of the event but I think we don't need now cause I have added condition if result is not passed then result will take as false